pennylane/transforms/metric_tensor.py

```diff
@@ -479,7 +479,7 @@ def _get_gen_op(op, allow_nonunitary, aux_wire):
 
     except KeyError as e:
         if allow_nonunitary:
-            mat = qml.matrix(qml.generator(op, format="observable"))
+            mat = qml.matrix(qml.generator(op)[0])
             return qml.ControlledQubitUnitary(mat, control_wires=aux_wire, wires=op.wires)

```

Comment:
Why not just `qml.ctrl(qml.generator(op)[0], control=aux_wire)`?

Reply:
Indeed! :100: This is a legacy line.

Reply:
In order to realize this, the `generator` of operations needs to be a queued object, in order for `ctrl` to lead to queuing as well. This change to the generators is left for a separate PR.